src/core/schema.js

```diff
@@ -0,0 +1,109 @@
+var debug = require('debug');
+var propertyTypes = require('./propertyTypes').propertyTypes;
+
+var warn = debug('core:schema:warn');
+
+/**
+ * A schema is classified as a schema for a single property if:
+ * - `type` is defined on the schema as a string.
+ * - OR `default` is defined on the schema, as a reserved keyword.
+ * - OR schema is empty.
+ */
+function isSingleProperty (schema) {
+  if ('type' in schema) {
+    return typeof schema.type === 'string';
+  }
+  return 'default' in schema || Object.keys(schema).length === 0;
+}
+module.exports.isSingleProperty = isSingleProperty;
+
+/**
+ * Build step to schema to use `type` to inject default value, parser, and stringifier.
+ *
+ * @param {object} schema
+ * @returns {object} Schema.
+ */
+module.exports.process = function (schema) {
+  // For single property schema, run processPropDefinition over the whole schema.
+  if (isSingleProperty(schema)) {
+    return processPropertyDefinition(schema);
+  }
+
+  // For multi-property schema, run processPropDefinition over each property definition.
+  Object.keys(schema).forEach(function (propName) {
+    schema[propName] = processPropertyDefinition(schema[propName]);
+  });
+  return schema;
+};
+
+/**
+ * Inject default value, parser, stringifier for single property.
+ */
+function processPropertyDefinition (propDefinition) {
+  var propType;
+  var defaultVal = propDefinition.default;
+  var typeName = propDefinition.type;
+
+  if (!propDefinition.type) {
+    if (defaultVal !== undefined && ['boolean', 'number'].indexOf(typeof defaultVal) !== -1) {
+      // Type inference.
+      typeName = typeof defaultVal;
+    } else {
+      // Fall back to string.
+      typeName = 'string';
+    }
+  } else if (propDefinition.type === 'bool') {
```

Comment:
No, `int` is a valid type that is different from `number`. `int` applies parseInt rather than parseFloat.